src/tools/rustbook/src/main.rs

```diff
@@ -46,23 +50,39 @@ fn main() {
 
                 ::std::process::exit(101);
             }
-        },
+        }
         ("linkcheck", Some(sub_matches)) => {
             if let Err(err) = linkcheck(sub_matches) {
                 eprintln!("Error: {}", err);
 
-                #[cfg(feature = "linkcheck")]
-                {
-                    if let Ok(broken_links) = err.downcast::<BrokenLinks>() {
-                        for cause in broken_links.links().iter() {
-                            eprintln!("\tCaused By: {}", cause);
-                        }
+                // HACK: ignore timeouts
+                let actually_broken = {
+                    #[cfg(feature = "linkcheck")]
+                    {
+                        err.downcast::<BrokenLinks>()
+                            .map(|broken_links| {
+                                broken_links
+                                    .links()
+                                    .iter()
+                                    .inspect(|cause| eprintln!("\tCaused By: {}", cause))
```

Comment:
This changes it so that only the first error will be displayed. Can it be changed so that it still displays all errors?

Also, it looks like 0.4 was released, which uses a significantly different API.  What is the plan to update that?

Reply:
> This changes it so that only the first error will be displayed. Can it be changed so that it still displays all errors?

Ah, good catch. Done.

> What is the plan to update that?

We haven't really updated yet on the rustc-guide repo. I need to take a look at the PR, but haven't had time. The main advantage would be the ability to take advantage of caching. I'm not sure that's a can of worms I want to open on this repo, though. It was hard enough to get what we have working...